src/main/java/org/eclipse/uprotocol/client/UMessageBuilder.java

```diff
@@ -0,0 +1,302 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.uprotocol.client;
+
+
+import org.eclipse.uprotocol.v1.*;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.Message;
+
+import org.eclipse.uprotocol.uuid.factory.UuidFactory;
+
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Builder for easy construction of the UAttributes object.
+ */
+public class UMessageBuilder {
+
+    private final UUri source;
+    private final UUID id;
+    private final UMessageType type;
+    private UPriority priority;
+    private Integer ttl;
+    private String token;
+    private UUri sink;
+    private Integer plevel;
+    private UCode commstatus;
+    private UUID reqid;
+    private String traceparent;
+
+    private UPayloadFormat format;
+    private ByteString payload;
+
+
+    /**
+     * Construct a UMessageBuilder for a publish message.
+     * @param source   Source address of the message.
+     * @return Returns the UMessageBuilder with the configured priority.
+     */
+    public static UMessageBuilder publish(UUri source) {
+        Objects.requireNonNull(source, "source cannot be null.");
+        return new UMessageBuilder(source, UuidFactory.Factories.UPROTOCOL.factory().create(), 
+        UMessageType.UMESSAGE_TYPE_PUBLISH);
+    }
+
+
+    /**
+     * Construct a UMessageBuilder for a notification message.
+     * @param source   Source address of the message.
+     * @param sink The destination URI.
+     * @param priority The priority of the message.
+     * @return Returns the UMessageBuilder with the configured priority and sink.
+     */
+    public static UMessageBuilder notification(UUri source, UUri sink) {
+        Objects.requireNonNull(source, "source cannot be null.");
```

Comment:
Gonna steal Pete's "nit", so this isn't breaking, but would be nice.

nit:

Would be good to not repeat the "source cannot be null.", "sink cannot be null." messages in each method... maybe make a variable or file with those error messages stored. This may not pass sonar testing